wolfssl/wolfcrypt/wc_port.h

```diff
@@ -1180,6 +1180,29 @@ WOLFSSL_ABI WOLFSSL_API int wolfCrypt_Cleanup(void);
     #endif
 #endif
 
+#ifdef WOLF_C99
```

Comment:
Looks like there are some more inline assembly variations we should support:

From sp_cortexm.c:

```
#ifdef __IAR_SYSTEMS_ICC__
#define __asm__        asm
#define __volatile__   volatile
#define WOLFSSL_NO_VAR_ASSIGN_REG
#endif /* __IAR_SYSTEMS_ICC__ */
#ifdef __KEIL__
#define __asm__        __asm
#define __volatile__   volatile
#endif /* __KEIL__ */
```

I think you might consider:

```
#ifdef WOLF_C99
    /* use alternate keyword for compatibility with -std=c99 */
    #define XASM_VOLATILE(a) __asm__ volatile(a)
#elif defined(__IAR_SYSTEMS_ICC__)
    #define XASM_VOLATILE(a) asm volatile(a)
#elif defined(__KEIL__)
    #define XASM_VOLATILE(a) __asm volatile(a)
#else
    #define XASM_VOLATILE(a) __asm__ __volatile__(a)
#endif
```